src/wp-includes/shortcodes.php

```diff
@@ -221,6 +221,14 @@ function do_shortcode( $content, $ignore_html = false ) {
 		return $content;
 	}
 
+	// Ensure this context is only added once if shortcodes are nested.
+	$has_filter = has_filter( 'wp_get_attachment_image_context', '_filter_do_shortcode_context' );
+	$filter_added = false;
+
+	if ( ! $has_filter ) {
+		$filter_added = add_filter( 'wp_get_attachment_image_context', '_filter_do_shortcode_context' );
+	}
```

Comment:
```suggestion
	$has_filter = has_filter( 'wp_get_attachment_image_context', '_filter_do_shortcode_context' );
	
	if ( ! $has_filter ) {
		$has_filter = add_filter( 'wp_get_attachment_image_context', '_filter_do_shortcode_context' );
	}
```

This is a little easier to read.

Reply:
@spacedmonkey This will lead to different behavior though. We need a separate `$filter_added` variable since the filter should only be removed further down if it was just added in the same execution. For example, we don't want to remove the filter callback if it was added outside of the current function execution.